### DISCUSSION TITLE
Port Forwarding Problem - Mullvad

### DISCUSSION THREAD

**Discussion by ub46692** [qdm12/gluetun#545] [Q&A]
Hello, I'm hoping that someone can assist me with a port forwarding issue that I am having. I'm sure that it's a configuration issue on my side that I am simply missing.

The Docker container is running on a Synology NAS. My VPN provider is Mullvad.

I have setup gluetun and here are some of the parameters I have been testing with:

CITY - Two different servers, one US, one Sweden
SERVER_HOSTNAME - Two different servers, one US, one Sweden
PORT_FORWARDING (tested with both ON and OFF)
FIREWALL (tested with both ON and OFF)
FIREWALL_VPN_INPUT_PORTS, two different ports. One for each server.

Note that I am only setting one server and one port at a time. I tested with two different servers to see if it was a server issue. 

I can post logs if necessary but I don't see any errors in the log. The container is assigned an address from Mullvad (I obfuscated for posting). 

```
2021/07/27 21:55:43 INFO dns over tls: ready
2021/07/27 21:55:45 INFO You are running on the bleeding edge of latest!
2021/07/27 21:55:47 INFO ip getter: Public IP address is xxx.xx.xxx.xxx (Sweden, Stockholm, Stockholm)
```

I checked the IP address in the container (via a shell) and it is the VPN Address. I have also forwarded the VPN port in my router to the Synology.

I setup the port checker and am not seeing any hits and the browser fails to connect as well. 

```
######### Port Checker ##########
######## by Quentin McGaw #######
######## Give some ?? at #########
# github.com/qdm12/port-checker #
#################################

2021/07/27 08:13:47 INFO listening on 0.0.0.0:57361
```

Any suggestions on what I can check or try next? 

Thank you,

Erick

**Comment by qdm12** [comment]
VPN port forwarding means you can access it through http://vpnip:forwardedport you can't access it on your LAN or router.

Try with `http://<vpn-ip>:57361` in your browser?

**Reply by ub46692** [reply]
Hi, thanks for the quick reply. 

Yes, I tried `http://<vpn-ip>:57361` from my non-vpn desktop web browser and no luck.

The browser displays:
```
This site can’t be reached xxx.xx.xxx.xxx refused to connect.
ERR_CONNECTION_REFUSED
```

vpn-ip being the same IP as displayed in the log entry "`2021/07/27 21:55:47 INFO ip getter: Public IP address is xxx.xx.xxx.xxx (Sweden, Stockholm, Stockholm).`"

I also tried a telnet to xxx.xx.xxx.xxx 57361 and no luck. 

Thanks,

Erick

**Reply by qdm12** [reply]
My provider being Mullvad as well, I'll test it tomorrow, maybe it's a bug on my side since I changed quite a lot of code recently.

Try maybe using Docker image tag `:v3.20.0`?

**Reply by ub46692** [reply]
OK. Will do. I will check it later tonight and share the results.

Thanks again for the quick replies and the container itself. Looking forward to getting it running and using it.

Erick

**Reply by ub46692** [reply]
I tried 3.20.0 tonight. No luck. I'm sure that I am doing something wrong on my side. 

I looked at the logs a bit more in depth and noticed some items related to the TUN interface.

Does this look OK? At first it looks like the TUN is not available but then later in the logs it seems to be running.

```
2021/07/28 05:24:59 INFO openvpn configurator: checking for device /dev/net/tun
2021/07/28 05:24:59 WARN TUN device is not available: open /dev/net/tun: no such file or directory
2021/07/28 05:24:59 INFO openvpn configurator: creating /dev/net/tun
2021/07/28 05:24:59 INFO firewall: firewall disabled, only updating allowed ports internal state
2021/07/28 05:25:01 INFO openvpn: setsockopt TCP_NODELAY=1 failed
2021/07/28 05:25:01 INFO openvpn: TUN/TAP device tun0 opened
2021/07/28 05:25:01 INFO openvpn: /sbin/ip link set dev tun0 up mtu 1500
2021/07/28 05:25:01 INFO openvpn: /sbin/ip link set dev tun0 up
2021/07/28 05:25:01 INFO openvpn: /sbin/ip addr add dev tun0 xx.x.x.xx/16
2021/07/28 05:25:01 INFO openvpn: Initialization Sequence Completed
2021/07/28 05:25:03 INFO dns over tls: ready
2021/07/28 05:25:03 INFO You are running the latest release v3.20.0
2021/07/28 05:25:03 INFO ip getter: Public IP address is xxx.xxx.xx.xx (United States, California, Long Beach)
```

Thank you,

Erick

**Reply by qdm12** [reply]
I just tested on `:latest` and it still works, I think it's due to a configuration issue you are having.

My public ip was `68.235.43.21` and port forwarded was `9876`

My command was:

```sh
docker run -it --rm --cap-add=NET_ADMIN -e VPNSP=mullvad -e USER=someNumber --name=gluetun -e SERVER_HOSTNAME=us-chi-003 -e FIREWALL_VPN_INPUT_PORTS=9876 qmcgaw/gluetun
```

Then in another shell:

```sh
wget -qO port-checker https://github.com/qdm12/port-checker/releases/download/v0.3.0/port-checker_0.3.0_linux_amd64
chmod +x port-checker
./port-checker -port 9876
```

Then in my browser accessing `http://68.235.43.21:9876` works.

Maybe that can help?

**Comment by ub46692** [comment]
Hi Quentin,

Thank you *very much* for the information. I wrote up a reply last night
(when I tested 3.20.0) but it looks like I didn't submit it. I didn't have
any luck with that version either.

I took your Docker statement and modified it for my configuration.

```
docker run --rm --cap-add=NET_ADMIN \
-e VPNSP=mullvad \
-e USER=accountnumber\
--name=gluetun \
-e SERVER_HOSTNAME=us-phx-004 \
-e FIREWALL_VPN_INPUT_PORTS=56386 \
-e PUID=1027 \
-e PGID=100 \
-v /volume1/docker/gluetun/config:/gluetun \
qmcgaw/gluetun
```

There were some differences from what I was using before. I was using
*OPENVPN_USER* instead of* USER*. I was also setting *OPENVPN_PASSWORD *to*
m *as I had seen on several posts.

Unfortunately, I still was unable to connect.

[image: image.png]

[image: image.png]

Port forwarding rule

[image: image.png]

I attached my gluetun run log.

Thank you again for your help with this. I do appreciate it. I am going to
keep trying. It's definitely something on my end.

Thanks,

Erick

**Reply by qdm12** [reply]
> I looked at the logs a bit more in depth and noticed some items related to the TUN interface.

That's all good. Gluetun manages to create it. If you want you can map the /dev/net/tun device but it doesn't really change anything. Soon Gluetun will create wireguard devices too, so no need to worry on that one 😉 I might remove the warning as  it's misleading today.

`USER` is the old variable I still use for dev for some reason I guess because it's shorter.

Setting an openvpn password for Mullvad is ignored anyway since there isn't one.

Are you sure you use the same SERVER_HOSTNAME as your port? Since now Mullvad forwards a port for a particular server location, you need the hostname to match that location.

**Comment by ub46692** [comment]
Hi Quentin,

Thank you so much for your detailed reply and assistance. I appreciate it.

So I found a couple of mistakes on my side.

1) I reread the Mullvad port forwarding page and something glaring jumped
out at me

[image: image.png]

I had mistakenly chosen a key to associate with it and I shouldn't have.
Because of OpenVPN, I should have selected "No Key".

[image: image.png]

2) I still was having an issue after this and I found one other problem
(and this one is *really dumb*, my apologies for wasting your time).

I was starting port checker with an SSH shell to my Synology itself.

It then all of a sudden hit me that it was not running within the container
itself and of course it would not be accessible from the outside world.

As soon as I launched a shell within gluetun and ran port checker there,
all was good!!!

Very lame on my part. Sorry about wasting your time on that. Because of
that, I sent you some money via Paypal to go buy yourself a beer or
something.

[image: image.png]

Thanks again. Now on to the fun stuff!!

Appreciate it. Have a good day.

Erick

**Reply by qdm12** [reply]
Hi Erik, thanks for the donation!

Happy to help, these things are complex, even for myself 😄

I'm glad you figured it out 🎉
